src/Cli.Tests/ConfigGeneratorTests.cs

```diff
@@ -119,6 +121,73 @@ public void TestCleanup()
         }
     }
 
+    /// <summary>
+    /// Test to verify creation of initial config with special characters
+    /// such as [!,@,#,$,%,^,&,*, ,(,)] in connection-string or graphql api
+    /// </summary>
+    [TestMethod]
+    public void TestSpecialCharactersInConnectionString()
+    {
+        HandleConfigFileCreationAndDeletion(TEST_RUNTIME_CONFIG_FILE, configFilePresent: false);
+        InitOptions options = new(
+            databaseType: DatabaseType.MSSQL,
+            connectionString: "A!string@with#some$special%characters^to&check*proper(serialization).'",
+            cosmosNoSqlDatabase: null,
+            cosmosNoSqlContainer: null,
+            graphQLSchemaPath: null,
+            graphQLPath: "/An_",
+            setSessionContext: false,
+            hostMode: HostMode.Production,
+            corsOrigin: null,
+            authenticationProvider: EasyAuthType.StaticWebApps.ToString(),
+            config: TEST_RUNTIME_CONFIG_FILE);
+
+        StringBuilder expectedRuntimeConfigJson = new(
+        @"{" +
+            @"""$schema"": """ + DAB_DRAFT_SCHEMA_TEST_PATH + @"""" + "," +
+            @"""data-source"": {
+                    ""database-type"": ""mssql"",
+                    ""connection-string"": ""A!string@with#some$special%characters^to&check*proper(serialization).'"",
+                    ""options"":{
+                        ""set-session-context"": false
+                    }
+                },
+           ""runtime"": {
+                ""rest"": {
+                  ""enabled"": true,
+                  ""path"": ""/api""
+                  },
+                ""graphql"": {
+                  ""enabled"": true,
+                  ""path"": ""/An_"",
+                  ""allow-introspection"": true
+                  },
+                ""host"": {
+                  ""cors"": {
+                    ""origins"": [],
+                    ""allow-credentials"": false
+                        },
+                  ""authentication"": {
+                    ""provider"": ""StaticWebApps""
+                        },
+                  ""mode"": ""production""
+                  }
+              },
+              ""entities"": {}
+            }");
+
+        expectedRuntimeConfigJson = expectedRuntimeConfigJson.Replace("\n", string.Empty);
+        expectedRuntimeConfigJson = expectedRuntimeConfigJson.Replace(" ", string.Empty);
+
+        Assert.IsTrue(TryGenerateConfig(options, _runtimeConfigLoader!, _fileSystem!));
+
+        StringBuilder actualRuntimeConfigJson = new(_fileSystem!.File.ReadAllText(TEST_RUNTIME_CONFIG_FILE, Encoding.Default));
+        actualRuntimeConfigJson = actualRuntimeConfigJson.Replace(" ", string.Empty);
+        actualRuntimeConfigJson = actualRuntimeConfigJson.Replace("\r\n", string.Empty);
```

Comment:
Should expected runtime config also replace `"\r\n"`?

Reply:
Comparing JSON as a string can be problematic because it does not take into account differences in formatting or ordering of the JSON properties. This can lead to false negatives in tests. I have updated it to use JToken.DeepEquals,  it compares the JSON objects in a more robust way that takes into account differences in formatting and ordering.

Reply:
Parsing it as JSON objects loses the test scenario - as I mentioned in the PR description as well:

> Note, this test is modified to explicitly compare the expected and actual json string from file since any usage of JsonSerializer/JObject.Parse would require using the same encoder to deserialize making the test itself run similar code which it is testing.


If the test failing, replacing `\r\n` might also be needed to consider platform differences.